src/main/java/com/couchbase/lite/replicator/ChangeTracker.java

```diff
@@ -630,10 +644,11 @@ public void setPaused(boolean paused) {
 
     protected void waitIfPaused(){
         synchronized (pausedObj) {
-            while (paused) {
+            while (paused && running) {
```

Comment:
@zgramana suggested `Paused` state should be one of StateMachine state. We should discuss about it. Ticket for it. https://github.com/couchbase/couchbase-lite-java-core/issues/1045

I will notify pauseObj when state changed